index.js

```diff
@@ -164,17 +167,21 @@ Warehouse.prototype.makeRequest = function makeRequest(uri, pathname, options, n
       || err.message.includes('400');
   });
 
-  operation.attempt(fn => {
-    request(options, function replied(error, response, body) {
-      if (error) return fn(error);
+  operation.attempt(async fn => {
+    try {
+      const res = await fetch(options.url, options);
+      const body = await res.json();
```

Comment:
This should probably move to after the status code checks

Reply:
oh, i see, they're used in the error messages. hmm. if there was no json or the response wasn't json, this is probably going to throw, but instead of getting a good error with the status code, you're going to end up with an error about the bad json parse.